internal/controller/controllers/agent_controller.go

```diff
@@ -487,9 +488,11 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 			ret = ctrl.Result{RequeueAfter: r.ApproveCsrsRequeueDuration}
 		}
 	}
-	if updateErr := r.Status().Update(ctx, agent); updateErr != nil {
-		log.WithError(updateErr).Error("failed to update agent Status")
-		return ctrl.Result{Requeue: true}, nil
+	if !reflect.DeepEqual(agent, origAgent) {
```

Comment:
I suggest that we'll add an `else` here with a debug print, to indicate that no updates were found and thus `r.Status().Update()` was not invoked.
Otherwise, it would be hard to debug possible situations where we expected an update that didn't happen.

Reply:
Also, this should be covered by tests.

Reply:
OK